client/my-sites/patterns/index.node.tsx

```diff
@@ -22,15 +25,19 @@ function renderPatterns( context: RouterContext, next: RouterNext ) {
 }
 
 function fetchCategoriesAndPatterns( context: RouterContext, next: RouterNext ) {
+	performanceMark( context, 'fetchCategoriesAndPatterns' );
+
 	const { cachedMarkup, queryClient, lang, params, store } = context;
 
-	if ( cachedMarkup ) {
+	if ( cachedMarkup || Object.keys( context.query ).length > 0 ) {
```

Comment:
I think it would be useful to clarify the intent of this condition, e.g.:

```suggestion
	// Bypasses fetching if the content is already cached, or if some query parameters were specified in the url
	if ( cachedMarkup || Object.keys( context.query ).length > 0 ) {
```